Distance_2/include/CGAL/squared_distance_2_1.h

```diff
@@ -197,7 +197,109 @@ namespace internal {
   typename K::FT
   squared_distance(const typename K::Segment_2 &seg1,
 		   const typename K::Segment_2 &seg2,
-		   const K& k)
+		   const K& k,
+                   const Cartesian_tag&)
```

Comment:
Is there still a need to disjoint Cartesian and Homogeneous kernels?

Reply:
I don't want to work on Homogeneous.

Reply:
My point is that the code for the Cartesian kernel doesn't seem to be specific to a Cartesian kernel anymore (?)

Reply:
I agree with Mael, here: it looks like the the code with `Cartesian_tag` could be applied to the homogeneous case as well. I also think we could merge the two overloads of `squared_distance(Segment_2, Segment_2, K, Tag)` into one.